08-mapping.Rmd

```diff
@@ -53,24 +53,22 @@ Finally, Section \@ref(other-mapping-packages) covers a range of alternative map
-Things have advanced greatly since **sp** was released [see @pebesma_classes_2005].
-Many new techniques for map making have been developed since then.
-However, a decade later static plotting was still the emphasis of geographic data visualisation in R [@cheshire_spatial_2015].
-
-Despite the innovation of interactive mapping in R, static maps are still the foundation of mapping in R.
-The generic `plot()` function is often the fastest way to create static maps from vector and raster spatial objects, as shown in Sections \@ref(basic-map) and \@ref(basic-map-raster).
-Sometimes simplicity and speed are priorities, especially during the development phase of a project, and this is where `plot()` excels.
+Standard formats include `.png` and `.pdf` for raster and vector outputs respectively.
+Initially, static maps were the only type of maps that R could produce.
+Things have advanced greatly since the release of **sp** [see @pebesma_classes_2005] and many new techniques for map making have been developed.
+However, a decade later static plotting was still the emphasis of geographic data visualisation in R [@cheshire_spatial_2015] and still is its foundation despite the innovation of interactive mapping.
+
+The generic `plot()` function is often the fastest way to create static maps from vector and raster spatial objects (see sections \@ref(basic-map) and \@ref(basic-map-raster)).
+Sometimes, simplicity and speed are priorities, especially during the development phase of a project, and this is where `plot()` excels.
 The base R approach is also extensible, with `plot()` offering dozens of arguments.
-Another low-level approach is the **grid** package, which provides functions for low-level control of graphical outputs, --- see *R Graphics* [@murrell_r_2016], especially  Chapter [14](https://www.stat.auckland.ac.nz/~paul/RG2e/chapter14.html).
-The focus of this section, however, is making static maps with **tmap**, emphasizing the important aesthetic and layout options.
-
-Why **tmap**?
-It is a powerful and flexible map-making package with sensible defaults.
-It has a concise syntax that allows for the creation of attractive maps with minimal code, which will be familiar to **ggplot2** users.
-Furthermore, **tmap** has a unique capability to generate static and interactive maps using the same code via `tmap_mode()`.
-It accepts a wider range of spatial classes (including `raster` objects) than alternatives such as **ggplot2**, as documented in vignettes [`tmap-getstarted`](https://cran.r-project.org/web/packages/tmap/vignettes/tmap-getstarted.html) and [`tmap-changes-v2`](https://cran.r-project.org/web/packages/tmap/vignettes/tmap-changes-v2.html) and an academic paper on the subject [@tennekes_tmap_2018].
+Another low-level approach is the **grid** package which provides functions for basic control of graphical outputs---see *R Graphics* [@murrell_r_2016], especially  Chapter [14](https://www.stat.auckland.ac.nz/~paul/RG2e/chapter14.html). 
```

Comment:
You've changed the style of the 'em dash' so there is no space. That seems to be common stylistic practice: https://www.quickanddirtytips.com/education/grammar/em-dash-space . However, it's not in the AP style guide (we didn't follow this, but it's popular): https://english.stackexchange.com/questions/214585/space-before-after-emdash-or-not

I'm not against this em dash change. However, I think that if it's done here, it needs to be done everywhere, including these locations, found by [ripgrep](https://github.com/BurntSushi/ripgrep#why-should-i-use-ripgrep): 

```
rg -tr ' --- '
11-spatial-cv.Rmd
33:Supervised techniques model the relationship between such responses --- which are known for a sample of observations --- and one or more predictors.

03-attribute-operations.Rmd
116:You use indices to specify the elements you wish to extract from an object, e.g., `object[i, j]`, with `i` and `j` typically being numbers or logical vectors --- `TRUE`s and `FALSE`s --- representing rows and columns (they can also be character strings, indicating row or column names).
377:**dplyr** has multiple join functions including `left_join()` and `inner_join()` --- see `vignette("two-table")` for a full list.
661:For example, `stack(elev, grain)[1]` returns a matrix with one row and two columns --- one for each layer.

15-synthesis.Rmd
76:While commonly needed operators/functions were covered in depth --- such as the base R `[` subsetting operator and the **dplyr** function `filter()` --- there are many other functions for working with geographic data, from other packages, that have not been mentioned.
186:that could supplement/replace the **raster** package used in this book --- see @garrard_geoprocessing_2016 and online tutorials such as [automating-gis-processes](https://automating-gis-processes.github.io/CSC18) for more on the Python ecosystem.
224:Both capture the essence of working with geographic data, but geocomputation has advantages: it concisely captures the 'computational' way of working with geographic data advocated in this book --- implemented in code and therefore encouraging reproducibility --- and builds on desirable ingredients of its early definition [@openshaw_geocomputation_2000]:

08-mapping.Rmd
30:Map making --- the art of cartography --- is an ancient skill that involves communication, intuition, and an element of creativity.
66:Another low-level approach is the **grid** package, which provides functions for low-level control of graphical outputs, --- see *R Graphics* [@murrell_r_2016], especially  Chapter [14](https://www.stat.auckland.ac.nz/~paul/RG2e/chapter14.html).
270:Although `style` is an argument of **tmap** functions, in fact it originates as an argument in `classInt::classIntervals()` --- see the help page of this function for details.
287:Sequential palettes can be single (`Blues` go from light to dark blue, for example) or multi-color/hue (`YlOrBr` is gradient from light yellow to brown via orange, for example), as demonstrated in the code chunk below --- output not shown, run the code yourself to see the results!
770:Each of these (particularly GeoServer) is scalable, enabling maps to be served to thousands of people daily --- assuming there is sufficient public interest in your maps!
791:The code below defines and launches --- with the command `shinyApp()` --- a lifeApp, which provides an interactive slider allowing users to make countries appear with progressively lower levels of life expectancy (see Figure \@ref(fig:lifeApp)):
815:This contains input and output 'widgets' --- in this case, a `sliderInput()` (many other `*Input()` functions are available) and a `leafletOutput()`.
819:`output` is a list of objects containing elements generated by `render*()` function --- `renderLeaflet()` which in this example generates `output$map`.
820:Input elements such as `input$life` referred to in the server must relate to elements that exist in the `ui` --- defined by `inputId = "life"` in the code above.
829:These include `reactive()` and `observe()` (for creating outputs that respond to the user interface --- see `?reactive`) and `leafletProxy()` (for modifying a `leaflet` object that has already been created).

05-geometry-operations.Rmd
...
```

Reply:
So I suggest you replace that with something like this, which is more concise and avoids the em dash altogether: 

```
Another low-level approach is the **grid** package which provides allows low level control of static maps, as illustrated in Chapter [14](https://www.stat.auckland.ac.nz/~paul/RG2e/chapter14.html) of @murrell_r_2016. 
```

Reply:
I actually also thought about that (that I was not following the style that you used throughout the book). I should have removed that wrong comma, but left the spaces.

 I can edit with your new suggestion or add the spaces (whichever you prefer): I am of the em-dash without space school, but can live with spaces around it too :slightly_smiling_face:

Reply:
I say go with the new 'no em dash' suggestion. I think my natural writing style could be described as 'comma sparse, em dash happy' (the proof readers added lots of the commas).

Reply:
> (the proof readers added lots of the commas)

I did notice that you write 'comma sparse[ly]' and am often tempted to add additional commas, even after the proof readers :slightly_smiling_face: That one though (before the em-dash), if it was added by a proof reader, was definitely wrong.

On my end, I am parenthesis heavy, which is the worse of all :laughing: